src/deploy/functions/prepare.ts

```diff
@@ -186,6 +187,7 @@ export function inferDetailsFromExisting(
   usedDotenv: boolean
 ): void {
   for (const wantE of backend.allEndpoints(want)) {
+    serviceForEndpoint(wantE).copyResourceOptionsToEndpoint(wantE as any, want);
```

Comment:
You're correct, this is pretty brittle. Maybe I should combine these into one function? I was trying to not iterate over the endpoints another time since we already do it quite a bit

Reply:
combined them into a single function that's invoked from the prepare stage